src/poliastro/atmosphere/jacchia.py

```diff
@@ -0,0 +1,339 @@
+            )
+            self.CM[iz] = self.CM[iz] * phid0
+
+        self.y = self.WM[150]
+        self.WM[150] = 0
+
+        for iz in range(151, alt):
+            self.x = self.WM[iz - 1] + (self.y + self.WM[iz])
+            self.y = self.WM[iz]
+            self.WM[iz] = self.x
+
+        for iz in range(150, alt):
+            self.WM[iz] = np.exp(self.WM[iz]) * (self.T[iz] / self.T[150]) ** 0.75
+            self.CM[iz] = self.WM[iz] * self.CM[iz]
+
+        self.y = self.CM[150]
+        self.CM[150] = 0
+
+        for iz in range(151, alt):
+            self.x = self.CM[iz - 1] + 0.5 * (self.y + self.CM[iz])
+            self.y = self.CM[iz]
+            self.CM[iz] = self.x
+
+        for iz in range(150, alt):
+            self.CH[iz] = (self.WM[500] / self.WM[iz]) * (
+                H_500 - (self.CM[iz] - self.CM[500])
+            )
+
+        for iz in range(150, alt):
+            self.CM[iz] = (
+                self.CN2[iz]
+                + self.CO2[iz]
+                + self.CO[iz]
+                + self.CAr[iz]
+                + self.CHe[iz]
+                + self.CH[iz]
+            )
+            self.WM[iz] = (
+                wmN2 * self.CN2[iz]
+                + wmO2 * self.CO2[iz]
+                + wmO * self.CO[iz]
+                + wmAr * self.CAr[iz]
+                + wmHe * self.CHe[iz]
+                + wmH * self.CH[iz]
+            ) / self.CM[iz]
+
+    def _O_and_O2_correction(self, alt, Texo):
+        """Add Jacchia 1977 empirical corrections to [O] and [O2]"""
+
+        for iz in range(90, alt):
+            self.CO2[iz] = self.CO2[iz] * (
+                10.0 ** (-0.07 * (1.0 + np.tanh(0.18 * (self.Z[iz] - 111.0))))
+            )
+            self.CO[iz] = self.CO[iz] * (
+                10.0 ** (-0.24 * np.exp(-0.009 * (self.Z[iz] - 97.7) ** 2))
+            )
+            self.CM[iz] = (
+                self.CN2[iz]
+                + self.CO2[iz]
+                + self.CO[iz]
+                + self.CAr[iz]
+                + self.CHe[iz]
+                + self.CH[iz]
+            )
+            self.WM[iz] = (
+                wmN2 * self.CN2[iz]
+                + wmO2 * self.CO2[iz]
+                + wmO * self.CO[iz]
+                + wmAr * self.CAr[iz]
+                + wmHe * self.CHe[iz]
+                + wmH * self.CH[iz]
+            ) / self.CM[iz]
+
+    def altitude_profile(self, alt, Texo):
+        """ Solves for atmospheric altitude profile at given altitude and exospheric temperature.
+
+        Parameters
+        ----------
+        alt: ~astropy.units.Quantity
+            Geometric/Geopotential altitude.
+        Texo: ~astropy.units.Quantity
+            Exospheric temperature
+
+        Returns
+        -------
+        altitude_profile: list
+            [altitude(Z), T, N2, O2, O, Ar, He, H, Total number density, Mean Molecular weight]
+        """
+
+        if 150 <= alt.value < 500:
```

Comment:
Altitude values should be previously "hard converted" to a desired standard units before dealing with their numerical value. If user passes a `120 * u.m` value obtained would be `120` and internal code would manage as kilometers. Please, **force the conversion** before retrieving value for internal use to ensure desired units :+1: